ga4gh/server/datamodel/variants.py

```diff
@@ -37,6 +38,34 @@ def isUnspecified(str):
     return str == "" or str is None
 
 
+def deepGetAttr(obj, path):
+    """
+    Utility function which is similiar to std lib's getattr, but works
+    to resolve a dot-delimited path on an object. If path is not found
+    an `AttributeError` will be raised.
+    """
+    return reduce(getattr, path.split('.'), obj)
+
+
+def deepSetAttr(obj, path, val):
```

Comment:
Cool. I'm moving these over to schemas then.

Here's the recursive version:

```python
def deepGetAttr(obj, path):
    first, sep, rest = path.partition(".")
    obj = getattr(obj, first)
    if rest:
        obj = deepGetAttr(obj, rest)
    return obj
```

Seems like the recursive version is a little bit uglier. Any specific benefit you were thinking of?